src/commands/executeFunction/eventGrid/OpenEventGridFileStep.ts

```diff
@@ -0,0 +1,85 @@
+import { AzExtFsExtra, AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import * as os from 'os';
+import * as path from "path";
+import * as vscode from 'vscode';
+import { type Progress } from "vscode";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { feedUtils } from "../../../utils/feedUtils";
+import { type ExecuteEventGridFunctionContext } from "./ExecuteEventGridFunctionContext";
+
+export class OpenEventGridFileStep extends AzureWizardExecuteStep<ExecuteEventGridFunctionContext> {
+    public priority: number;
+
+    public async execute(context: ExecuteEventGridFunctionContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined; }>): Promise<void> {
+        if (!context.eventSource || !context.selectedFileName || !context.selectedFileUrl) {
+            throw new Error('Internal error: event source or event type is missing');
+        }
+
+        // Get selected contents of sample request
+        const downloadingMsg: string = localize('downloadingSample', 'Downloading sample request...');
+        progress.report({ message: downloadingMsg });
+        context.selectedFileContent = await feedUtils.getJsonFeed(context, context.selectedFileUrl);
+
+        // Create a temp file with the sample request & open in new window
+        const openingFileMsg: string = localize('openingFile', 'Opening file...');
+        progress.report({ message: openingFileMsg });
+        const tempFilePath: string = await createTempSampleFile(context.eventSource, context.selectedFileName, context.selectedFileContent);
+        const document: vscode.TextDocument = await vscode.workspace.openTextDocument(tempFilePath);
+        await vscode.window.showTextDocument(document, {
+            preview: false,
+        });
+        ext.fileToFunctionNodeMap.set(document.fileName, nonNullProp(ext, 'currentExecutingFunctionNode'));
+        context.fileOpened = true;
+
+        // Request will be sent when the user clicks on the button or on the codelens link
+        const doneMsg = localize('modifyFile', "You can modify the file and then click the 'Save and execute' button to send the request.");
```

Comment:
I think this notification is helpful, but it seems like it could get annoying if you were executing your trigger multiple times.

Maybe we could either a.) have a "don't show again" button or b.) have it timed to only show once a session or something like that

Reply:
Made it such that I set a flag on `ext.context.workspaceState` so that it shows only once per workspace. I couldn't find a way to make it show only once per _session_, but let me know if you do! (I guess I could reset the flag to `false` in the `activate` function?)

Reply:
I think that will be helpful enough!